swap/src/asb/event_loop.rs

```diff
@@ -318,16 +318,30 @@ where
         &mut self,
         min_buy: bitcoin::Amount,
         max_buy: bitcoin::Amount,
+        monero_wallet: Arc<monero::Wallet>,
     ) -> Result<BidQuote> {
+        let max_bitcoin_for_monero: bitcoin::Amount;
+        let mut max_bitcoin_buy = max_buy;
+        let monero_balance = monero_wallet.get_balance().await?.as_monero();
+
         let rate = self
             .latest_rate
             .latest_rate()
             .context("Failed to get latest rate")?;
 
+        max_bitcoin_for_monero = bitcoin::Amount::from_btc(monero_balance * rate.ask()?.as_btc())?;
+        if max_buy > max_bitcoin_for_monero {
+            tracing::warn!(
+                    "Your Max bitcoin to sell: {} is smaller than your current balance of monero available to sell {}",
+                    max_buy, max_bitcoin_for_monero
+                );
+            max_bitcoin_buy = max_bitcoin_for_monero
+        }
+
         Ok(BidQuote {
             price: rate.ask().context("Failed to compute asking price")?,
             min_quantity: min_buy,
```

Comment:
If `max_bitcoin_buy` is less than `min_buy`, the user cannot initiate a swap at all. We need to handle this case somehow. I suggest that we set both `min_quantity` and `max_quantity` to zero to communicate to the user that the ASB is not accepting swaps at the moment.

Reply:
I think we could make another issue for the project so we can solve it in parts

Reply:
I don't think we need another PR for this. Afaik this is already done for when the actual swap is started. You just need to subtract the fee from the Monero balance when calculating the `max_quantity`

Reply:
You are right, I have implemented a the calculation to subtract the fee

Reply:
Thanks a lot for implementing this fee subtraction thing, looks good to me!

The scenario I originally mentioned where the min_quantity is bigger than the max_quantity is still possible afaik (mistake on my end). I'd suggest than we just check if min_quantity > max_quantity and then return a Quote where both are zero.

If either of you has a better suggestion, I'm definitely open to it.